core/io/resource_loader.cpp

```diff
@@ -981,17 +981,17 @@ String ResourceLoader::path_remap(const String &p_path) {
 }
 
 void ResourceLoader::reload_translation_remaps() {
-	ResourceCache::lock.lock();
-
 	List<Resource *> to_reload;
-	SelfList<Resource> *E = remapped_list.first();
 
-	while (E) {
-		to_reload.push_back(E->self());
-		E = E->next();
-	}
+	{
+		MutexLock locked(ResourceCache::resources_mutex);
```

Comment:
@AThousandShips I've preserved the previous behavior here (only using MutexLock instead of lock/unlock), but I don't immediately see why this code needs to lock the resources_mutex at all. `remapped_list` is not under that guard anywhere else. ResourceCache explicitly friend's ResourceLoader so it can have access to internals. Wondering if this is vestigial cruft that ought to be removed?